satpy/readers/abi_l1b.py

```diff
@@ -53,14 +53,33 @@ def __init__(self, filename, filename_info, filetype_info):
         # xarray's default netcdf4 engine
         self.nc = xr.open_dataset(filename,
                                   decode_cf=True,
-                                  mask_and_scale=True,
+                                  mask_and_scale=False,
                                   chunks={'x': CHUNK_SIZE, 'y': CHUNK_SIZE})
         self.nc = self.nc.rename({'t': 'time'})
         platform_shortname = filename_info['platform_shortname']
         self.platform_name = PLATFORM_NAMES.get(platform_shortname)
         self.sensor = 'abi'
         self.nlines, self.ncols = self.nc["Rad"].shape
 
+    def __getitem__(self, item):
+        """Wrapper around `self.nc[item]`.
+
+        Some datasets use a 32-bit float scaling factor like the 'x' and 'y'
+        variables which causes inaccurate unscaled data values. This method
+        forces the scale factor to a 64-bit float first.
+
+        """
+        data = self.nc[item]
+        factor = data.attrs.get('scale_factor')
+        offset = data.attrs.get('add_offset')
+        fill = data.attrs.get('_FillValue')
```

Comment:
This is just for ABI L1B. I haven't seen a case where those are used. It could always be modified, but I'm not sure it is needed. I'm sure I'll regret not doing it in the future, but I also am not sure I want things to break if I do it now.